pkg/steps/template.go

```diff
@@ -59,6 +65,30 @@ func (s *templateExecutionStep) Run(ctx context.Context, dry bool) error {
 		return fmt.Errorf("template %s has no objects", s.template.Name)
 	}
 
+	if s.resources != nil {
+		containerResources, err := resourcesFor(s.resources.RequirementsForStep(s.template.Name))
+		if err != nil {
+			return fmt.Errorf("unable to calculate resources for %s: %s", s.template.Name, err)
+		}
+
+		for _, object := range s.template.Objects {
+			var pod *coreapi.Pod
+			yaml.Unmarshal(object.Raw, &pod)
+
+			if pod != nil {
```

Comment:
You can only do this for things that are pods. This doesn't check that things are pods.

Reply:
What's the type to deserialize into that has GVR ?

Reply:
My intention here was to check if the `Unmarshal` didn't work because the object can be different than `pod` and the pod var will continue to be `nil`. In this case, we don't want to continue.

Reply:
Also, the Unmarshal won't raise an error if the object is different than `pod`.

Reply:
Easy to do something like this, though: https://github.com/openshift/release/blob/297849cc49ed9c4d1ba9ae94c5104f36653f1ec8/tools/release-ci/pkg/buildrunner/runbuild.go#L65-L72